### PR TITLE
added shopping cart

### PR DESCRIPTION
**Pull request by dmytrocherepov**

### CODE REVIEW

File: src/main/java/com/example/bookstoreapp/dto/shoppingcart/CartItemUpdateDto.java
Comment (viacheslav-torbin) [inline src/main/java/com/example/bookstoreapp/dto/shoppingcart/CartItemUpdateDto.java:1]: You use 2 different styles for records, pls pick one

File: src/main/java/com/example/bookstoreapp/dto/shoppingcart/CartItemUpdateDto.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.example.bookstoreapp.dto.shoppingcart;
+
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Positive;
+
+public record CartItemUpdateDto(
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstoreapp/dto/shoppingcart/CartItemUpdateDto.java]: A little bit confusing naming. Is it request or Response? 
CartItemUpdateRequestDto

File: src/main/java/com/example/bookstoreapp/dto/shoppingcart/CartDto.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.example.bookstoreapp.dto.shoppingcart;
+
+import java.util.List;
+
+public record CartDto(
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstoreapp/dto/shoppingcart/CartDto.java]: ShoppingCartDto

File: src/main/java/com/example/bookstoreapp/service/ShoppingCartService.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package com.example.bookstoreapp.service;
+
+import com.example.bookstoreapp.dto.shoppingcart.CartDto;
+import com.example.bookstoreapp.dto.shoppingcart.CartItemDto;
+import com.example.bookstoreapp.dto.shoppingcart.CartItemRequestDto;
+import com.example.bookstoreapp.dto.shoppingcart.CartItemUpdateDto;
+import org.springframework.stereotype.Service;
+
+@Service
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstoreapp/service/ShoppingCartService.java]: It's redundant annotation here

File: src/main/java/com/example/bookstoreapp/service/impl/CategoryServiceImpl.java [hunk 4]
```diff
@@ -37,9 +36,6 @@ public CategoryResponseDto getById(Long id) {
     @Override
     public CategoryResponseDto save(CategoryRequestDto categoryDto) {
         Category entity = categoryMapper.toEntity(categoryDto);
-        if (categoryRepository.existsByName(categoryDto.name())) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstoreapp/service/impl/CategoryServiceImpl.java:40]: I didn't find any code related to Category. Remove it from this PR

File: src/main/java/com/example/bookstoreapp/service/impl/ShoppingCartServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    @Override
+    public CartDto getCart() {
+        return shoppingCartMapper.toDto(getUserCart());
+    }
+
+    @Override
+    public CartItemDto addCartItem(CartItemRequestDto requestDto) {
+        CartItem cartItem = new CartItem();
+        cartItem.setQuantity(requestDto.quantity());
+        cartItem.setBook(bookRepository.findById(requestDto.bookId()).orElseThrow(
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstoreapp/service/impl/ShoppingCartServiceImpl.java]: It's better to create local variable for Book. Code is difficult to read

File: src/main/java/com/example/bookstoreapp/service/impl/ShoppingCartServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    public CartItemDto updateCartItem(CartItemUpdateDto updateDto, Long itemId) {
+        ShoppingCart userCart = getUserCart();
+        CartItem cartItem = cartItemRepository
+                .findCartItemByIdAndShoppingCart(itemId, userCart)
+                .orElseThrow(() -> new EntityNotFoundException("No such cartItem"));
+        cartItem.setQuantity(updateDto.quantity());
+        return cartItemMapper.toDto(cartItemRepository.save(cartItem));
+    }
+
+    @Override
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstoreapp/service/impl/ShoppingCartServiceImpl.java:66]: @Transactional

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution!

File: src/main/java/com/example/bookstoreapp/controller/AuthenticationController.java [hunk 7]
```diff
@@ -18,6 +19,7 @@
 @Tag(name = "User authorization/registration",
         description = "Endpoint for user registration and authorization")
 @RequiredArgsConstructor
+@Validated
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstoreapp/controller/AuthenticationController.java]: Why you added this annotation if you didn't use validation for PathVariable for example?

**Review by Elena-Bruyako — APPROVED** [review]
Good job! Only remove useless annotation!

**Pull request merged**